dht20/dht20.go

```diff
@@ -0,0 +1,103 @@
+// Package dht20 implements a driver for the DHT20 temperature and humidity sensor.
+//
+// Datasheet: https://cdn-shop.adafruit.com/product-files/5183/5193_DHT20.pdf
+
+package dht20
+
+import (
+	"time"
+
+	"tinygo.org/x/drivers"
+)
+
+// Device wraps an I2C connection to a DHT20 device.
+type Device struct {
+	bus            drivers.I2C
+	Address        uint16
+	data           [8]uint8
+	temperature    float32
+	humidity       float32
+	prevAccessTime time.Time
+}
+
+// New creates a new DHT20 connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus drivers.I2C) Device {
+	return Device{
+		bus:     bus,
+		Address: defaultAddress, // Using the address defined in registers.go
+	}
+}
+
+// Configure sets up the device for communication and initializes the registers if needed.
+func (d *Device) Configure() {
+	// Get the status word
+	d.data[0] = 0x71
+	d.bus.Tx(d.Address, d.data[:1], d.data[:1])
+	if d.data[0] != 0x18 {
+		// Initialize registers
+		d.initRegisters()
+	}
+	// Set the previous access time to the current time
+	d.prevAccessTime = time.Now()
+}
+
+// initRegisters initializes the registers 0x1B, 0x1C, and 0x1E to 0x00.
+func (d *Device) initRegisters() {
+	// Initialize register 0x1B
+	d.data[0] = 0x1B
+	d.data[1] = 0x00
+	d.bus.Tx(d.Address, d.data[:2], nil)
+
+	// Initialize register 0x1C
+	d.data[0] = 0x1C
+	d.data[1] = 0x00
+	d.bus.Tx(d.Address, d.data[:2], nil)
+
+	// Initialize register 0x1E
+	d.data[0] = 0x1E
+	d.data[1] = 0x00
+	d.bus.Tx(d.Address, d.data[:2], nil)
+}
+
+// Update reads data from the sensor and updates the temperature and humidity values.
+func (d *Device) Update(which drivers.Measurement) error {
+	// Check if 80ms have passed since the last access
+	if d.prevAccessTime.Add(80 * time.Millisecond).Before(time.Now()) {
```

Comment:
You can use `time.Since` instead `Add`, `Before` and `Now`.
Also, consider inverting the condition so the code is cleaner and with less nesting
```suggestion
	if time.Since(d.prevAccessTime) < 80 * time.Millisecond {
		return nil
	}
```

Reply:
Thank you for the good suggestion. I have made the correction.